tests/functional/s3-tagging.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python
+# coding=utf8
+
+from __future__ import print_function
+from six.moves.urllib.parse import urlencode
+
+import json
+import subprocess
+import time
+
+AWS = ["aws", "--endpoint-url", "http://localhost:5000", "s3api"]
+BUCKET = "test-%d" % int(time.time())
+
+
+def run_s3api(*params, **kwargs):
+    print(*params)
+    cmd = AWS + list(params)
+    try:
+        out = subprocess.check_output(cmd, stderr=subprocess.STDOUT)
+        # some s3api commands return nothing and json.loads
+        # doesn't like it
+        if out == '':
+            out = '{}'
+    except subprocess.CalledProcessError as exc:
+        print(exc.output)
+        raise
+    return json.loads(out)
+
+
+def run_create_object_with_tagging():
+    run_s3api("create-bucket", "--bucket", BUCKET)
+
+    tagging = {'unicode': '♜♞♝♛♚♝♞♜'}
+    run_s3api("put-object", "--bucket", BUCKET, "--key", "magic",
+              "--tagging", urlencode(tagging))
+    data = run_s3api("get-object-tagging", "--bucket", BUCKET,
+                     "--key", "magic")
+    assert data['TagSet'][0]['Value'] == tagging['unicode'].decode('utf8')
+
+    tagging = {'été': ''}
+    run_s3api("put-object", "--bucket", BUCKET, "--key", "magic",
+              "--tagging", urlencode(tagging))
+    data = run_s3api("get-object-tagging", "--bucket", BUCKET, "--key",
+                     "magic")
+    assert data['TagSet'][0]['Key'] == 'été'.decode('utf8')
```

Comment:
We cannot decode a string.
```
>>> 'été'.decode('utf8')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'str' object has no attribute 'decode'
```

Reply:
Writting test in shell is very prone to error IMO and more difficult to split, I'll however split mine in small parts

My bad, I didn't tested with python3